packages/client/graph-scaffolder/README.md

```diff
@@ -1,28 +1,129 @@
 # graph-scaffolder
-Given a layout/adapter function and data, graph-scaffolder creates the basic SVG structures and interaction hooks for interactive management of graph data.
+graph-scaffolder provides SVG/DOM scaffolding:
+- SVG/DOM hierarchies + data-binding via D3
+- Basic boiler-plate functions such as pan/zoom, highlighting
+- Hooks for building custom events
 
-graph-scaffolder is _not_ a layout engine, rather, it abstracts away boilder-plate code like zooming, panning, and provides a unified interface.
+Note: graph-scaffolder is _not_ a layout engine. It can leverage a layout function, provided that the algorithm produces output in the format below.
 
 
 ## Data format
-The graph data is specified in the format of `{nodes, edges}` and allows for nesting, for example:
+The graph data is specified in the format of `{nodes, edges}` and allows for nesting, nested nodes are expected to be positioend relative to their parent node.  For example:
+
 ```
-{
-	nodes: [
-		{
-		}
+const graphData = {
+  nodes: [
+    {
+      id: 'node1',
+      label: 'node1 label'
+      x: 100,
+      y: 100,
+      width: 100,
+      height: 100,
+      data: { value: 123 },
+      nodes: []
+    },
+    {
+      id: 'node2',
+      label: 'node2 label',
+      x: 400,
+      y: 400,
+      width: 100,
+      height: 100
+      data: { value: 234 },
+      nodes: [
+        {
+          id: 'nested node1',
+          label: 'nested'
+          x: 10,
+          y: 10,
+          width: 20,
+          height: 20,
+          data: { value: 999 },
+          nodes: []
+        }
+      ]
+    }
   ],
   edges: [
-    id: 'e1',
-    source: 'node1',
-    target: 'node2'
+    {
+      id: '1',
+      source: 'node1',
+      target: 'node2',
+      points: [ { x: 200, y: 150 }, { x: 400, y: 450 }],
+      data: { foo: 'bar' }
+    }
   ]
 }
 ```
 
+### Usage
+Simple renderer with Typescript
+
+```
+interface NodeData {
+  value: number;
+}
+
+interface EdgeData {
+  foo: string;
```

Comment:
You sure this should be `foo`?

Reply:
`NodeData` and `EdgeData` are arbitrary data structures, as in INode<NodeData> and IEdge<EdgeData> respectively, I should add some notes that it supports and curries the typing information.